libc/time/utimes.c

```diff
@@ -0,0 +1,528 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ * HEADER:
+ *    - sys/time.h
+ *
+ * TESTED:
+ *    - utimes()
+ *    - futimes()
+ *    - lutimes()
+ *
+ * Copyright 2022, 2023 Phoenix Systems
+ * Author: Ziemowit Leszczynski, Mateusz Bloch
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <unistd.h>
+#include <fcntl.h>
+#include <time.h>
+#include <errno.h>
+#include <string.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/time.h>
+
+#include "unity_fixture.h"
+
+
+static const char *const FILENAME = "utimes";
+static const char *const LINKNAME = "utimes_link";
+static const unsigned int LOOP_CNT = 10;
+
+
+typedef enum {
+	TEST_SET_NULL,
+	TEST_SET_NOW,
+	TEST_SET_PAST
+} test_t;
+
+
+typedef enum {
+	UTIMES,
+	FUTIMES,
+	LUTIMES
+} mode_test;
+
+
+void test_utimes(const char *filename, test_t test, mode_test mode)
+{
+	int fd;
+	struct timeval tv[2];
+	struct stat statbuf;
+	int64_t adiff_sec, mdiff_sec;
+/* FIXME: only second accuracy is supported */
+#ifndef __phoenix__
+	int64_t adiff_nsec, mdiff_nsec;
+#endif
+
+	if (test != TEST_SET_PAST) {
+		TEST_ASSERT_EQUAL_INT(0, gettimeofday(&tv[0], NULL));
+		tv[1] = tv[0];
+	}
+	else {
+		tv[0].tv_sec = random();
+		tv[1].tv_sec = random();
+		tv[0].tv_usec = random() % 1000;
+		tv[1].tv_usec = random() % 1000;
+	}
+
+	if (mode == UTIMES) {
+		TEST_ASSERT_EQUAL_INT(0, utimes(filename, test == TEST_SET_NULL ? NULL : tv));
```

Comment:
I think that it would also worth to add a message about test type in case of fail - now we wouldn't now it